pytest.warn doesn't appear to be working for deprecation warnings

I'm not sure what to make of this, but it appears to be failing on a very simple test:

```
(tempenv-6cb925871591c) ~/.v/tempenv-6cb925871591c $ cat t.py
import warnings

import pytest


def test_foo():
    with pytest.warns(PendingDeprecationWarning):
        warnings.warn("msg", PendingDeprecationWarning)
(tempenv-6cb925871591c) ~/.v/tempenv-6cb925871591c $ py.test t.py
====================================================== test session starts =======================================================
platform darwin -- Python 2.7.10, pytest-2.8.0, py-1.4.30, pluggy-0.3.1
rootdir: /Users/alex_gaynor/.virtualenvs/tempenv-6cb925871591c, inifile:
collected 1 items

t.py F

============================================================ FAILURES ============================================================
____________________________________________________________ test_foo ____________________________________________________________

    def test_foo():
        with pytest.warns(PendingDeprecationWarning):
>           warnings.warn("msg", PendingDeprecationWarning)
E           Failed: DID NOT WARN

t.py:8: Failed
==================================================== 1 failed in 0.01 seconds ====================================================
```

--- a/doc/en/recwarn.rst
+++ b/doc/en/recwarn.rst
@@ -41,6 +41,10 @@ additional information::
 Alternatively, you can examine raised warnings in detail using the
 :ref:`recwarn <recwarn>` fixture (see below).
 
+.. note::
+    ``DeprecationWarning`` and ``PendingDeprecationWarning`` are treated
+    differently; see :ref:`ensuring_function_triggers`.
+
 .. _recwarn:
 
 Recording warnings
@@ -87,6 +91,9 @@ Each recorded warning has the attributes ``message``, ``category``,
 class of the warning. The ``message`` is the warning itself; calling
 ``str(message)`` will return the actual message of the warning.
 
+.. note::
+    ``DeprecationWarning`` and ``PendingDeprecationWarning`` are treated
+    differently; see :ref:`ensuring_function_triggers`.
 
 .. _ensuring_function_triggers:
 
@@ -94,16 +101,17 @@ Ensuring a function triggers a deprecation warning
 -------------------------------------------------------
 
 You can also call a global helper for checking
-that a certain function call triggers a ``DeprecationWarning``::
+that a certain function call triggers a ``DeprecationWarning`` or
+``PendingDeprecationWarning``::
 
     import pytest
 
     def test_global():
         pytest.deprecated_call(myfunction, 17)
 
-By default, deprecation warnings will not be caught when using ``pytest.warns``
-or ``recwarn``, since the default Python warnings filters hide
-DeprecationWarnings. If you wish to record them in your own code, use the
+By default, ``DeprecationWarning`` and ``PendingDeprecationWarning`` will not be
+caught when using ``pytest.warns`` or ``recwarn`` because default Python warnings filters hide
+them. If you wish to record them in your own code, use the
 command ``warnings.simplefilter('always')``::
 
     import warnings

